src/lib.rs

```diff
@@ -201,6 +202,20 @@ impl Bibliography {
         biblatex
     }
 
+    /// Serialize this bibliography into a BibLaTeX string.
+    ///
+    /// `field_order` is something like `vec!["title", "author"]`. In that case, every entry in the
+    /// output String have the title and author fields first, and then the rest of the fields
+    /// oredered alphabetically.
+    pub fn to_biblatex_string_ordered(&self, field_order: Vec<&str>) -> String {
```

Comment:
Is it really important for you to provide the field order manually? If not, this is overkill.

The root of our problem is of course that `Entry::fields` is a `HashMap` which just returns its entries in random order. Changing it to a (identical in terms of interface) [`BTreeMap`](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html) will always return the elements in the natural ([`Ord`](https://doc.rust-lang.org/std/cmp/trait.Ord.html)) order of the keys without any extra infrastructure. The keys will thus always be returned in alphabetical order.

This is of course insufficient if you want a specific non-alphabetical order of keys. In this case, something like a field name struct wrapping a string could implement a custom `Ord`, putting common fields like `author`, `title`, and `journal` first and then deferring to the string `Ord` implementation for everything else would be in order. It would look something like this:

```rust
#[derive(Debug, Clone, Eq, PartialEq)]
#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
pub struct Entry {
    /// The citation key.
    pub key: String,
    /// Denotes the type of bibliography item (e.g. `Article`).
    pub entry_type: EntryType,
    /// Maps from field names to their associated chunk vectors.
    pub fields: BTreeMap<Key, Chunks>,
}

/// A field name for an entry field.
#[derive(Debug, Clone, Eq, PartialEq)]
pub struct Field(pub String);

impl Ord for Field {
    /* custom ord implementation here,
       putting author and title in front of all other things */
}
```

This way, the API would not fragment into various methods doing about the same thing with different arguments and likely, less code would be added as well. Additionally, I think that one canonical field order for biblatex is enough, it does not really need to be customizable.

A final pointer: If writing functions and methods, [prefer passing a slice](https://stackoverflow.com/a/24104029/1567835) (`&[&str]` in this case) instead of a vector reference. This gives the caller more flexibility.